specs/Standards/NonFungibleToken/Event.md

```diff
@@ -0,0 +1,107 @@
+# Non-Fungible Token Event
+
+Version `1.0.0`
+
+## Summary
+
+Standard interfaces for NFT contract actions.
+
+## Motivation
+
+NFT-driven apps perform many similar actions.
+For example - `minting`, `burning` and `transferring`.
+Each app may have their own way of performing these actions.
+This introduces inconsistency in capturing these events.
+This extension addresses that.
+
+It's common for NFT apps to transfer many or one token at a time.
+Other applications need to track these and similar events consistently.
+If not, tracking state across many NFT-driven apps become infeasible.
+
+We need a standard way to capture events.
+This was discussed here https://github.com/near/NEPs/issues/254.
+
+## Interface
+
+Many apps use different interfaces that represent the same action.
+This interface standardizes that process.
+It captures these actions through logs.
+
+```ts
+// Interface to capture an event
+// and return formatted event string.
+interface EventJsonLog {
+    // Takes `EventLogData` and returns
+    // `EVENT_JSON: <EVENT_LOG_DATA>`
+    (e:EventLogData):string
+}
+
+// Interface to capture data 
+// about an event
+// Arguments
+// * `standard`: name of standard e.g. nep171
+// * `version`: e.g. 1.0.0
+// * `event`: `nft_mint` | `nft_burn` | `nft_transfer`
+// * `data`: associate event data
+interface EventLogData {
+    standard:string,
+    version:string,
+    event:string,
+    data: NftMintLog[]|NftTransferLog[]|NftBurnLog[]
+}
+
+// An event log to capture token minting
+// Arguments
+// * `owner_id`: "account.near"
+// * `token_ids`: ["1", "abc"]
+// * `memo`: optional message
+interface NftMintLog {
+    owner_id:string,
+    token_ids:string[],
+    memo?:string
+}
+
+// An event log to capture token burning
+// Arguments
+// * `owner_id`: owner of tokens to burn
+// * `authorised_id`: approved account to burn
+// * `token_ids`: ["1","2"]
+// * `memo`: optional message
+interface NftBurnLog {
+    owner_id:string,
+    authorised_id:string,
+    token_ids:string[],
+    memo?:string
+}
+
+// An event log to capture token transfer
+// Arguments
+// * `authorised_id`: approved account to transfer
+// * `sender_id`: "account.near"
+// * `receiver_id`: "receiver.near"
+// * `token_ids`: ["1", "12345abc"]
+// * `memo`: optional message
+interface NftTransferLog {
+    authorised_id:string,
+    sender_id:string,
```

Comment:
I think @telezhnaya suggested:

```js
{
owner_id:"owner.near",
authorised_id:"authorised.sender.near",
receiver_id:"receiver.near",
token_ids: ["1"]
}
```

Though, if we change sender to owner we get a different sort of confusion, when `authorised_id` is omitted:

```js
{
owner_id:"owner.near",
receiver_id:"receiver.near",
token_ids: ["1"]
}
```

I don't have a suggestion so far, but I agree with (sender, authorised, receiver) being confusing since sender is not actually sending in this case, and (owner, receiver) is not immediately clear as well.

Reply:
```ts
{
old_owner_id:"owner.near",
authorised_id:"authorised.sender.near",
new_owner_id:"receiver.near",
token_ids: ["1"]
}
```

Makes sense. Though it won't have symmetrical interface with others. Is this ok?

Reply:
Can't answer on how `authorised_id` will be used but I can say all useful events in general e.g. `ApprovalEvent` you mentioned is part of the plan - standard nft events. 

I will be creating another pr shortly regarding and I invite to mention as many events you think should be standardised.

Reply:
thanks @frol makes sense!